src/test/java/com/mongodb/InterruptibilityItTest.java

```diff
@@ -0,0 +1,277 @@
+                + "        ],\n"
+                + "        'blockConnection': true,\n"
+                + "        'blockTimeMS': " + blockTimeout.toMillis() + "\n"
+                + "    }\n"
+                + "}"),
+                primary(client, testTimeout),
+                clientSettingsBuilderSupplier)) {
+            return assertThrows(expectedType, () ->
+                    inVirtualThread(() ->
+                            inInterruptedThread(interruptDelay, action))
+            );
+        }
+    }
+
+    private static void inVirtualThread(final Runnable action) throws TimeoutException {
+        try (ExecutorService ex = Executors.newVirtualThreadPerTaskExecutor()) {
+            try {
+                ex.submit(action).get(testTimeout.toNanos(), NANOSECONDS);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            } catch (ExecutionException e) {
+                Throwable cause = com.mongodb.assertions.Assertions.assertNotNull(e.getCause());
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                } else {
+                    throw new RuntimeException(cause);
+                }
+            }
+        }
+    }
+
+    private static void inInterruptedThread(final Duration interruptDelay, final Runnable action) {
+        Future<?> futureInterrupt;
+        if (interruptDelay.isZero()) {
+            Thread.currentThread().interrupt();
+            futureInterrupt = CompletableFuture.completedFuture(null);
+        } else {
+            Thread threadToInterrupt = Thread.currentThread();
+            futureInterrupt = ForkJoinPool.commonPool().submit(() -> {
+                try {
+                    Thread.sleep(interruptDelay);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException(e);
+                }
+                threadToInterrupt.interrupt();
+            });
+        }
+        try {
+            try {
+                action.run();
+            } finally {
+                // clear the interrupted status
+                // noinspection ResultOfMethodCallIgnored
+                Thread.interrupted();
+            }
+        } finally {
+            try {
+                futureInterrupt.get();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                //noinspection ThrowFromFinallyBlock
+                throw new RuntimeException(e);
+            } catch (ExecutionException e) {
+                //noinspection ThrowFromFinallyBlock
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private static ServerAddress primary(final MongoClient client, final Duration timeout) {
+        Timeout tOut = Timeout.startNow(timeout.toNanos());
```

Comment:
tOut => timeout

(avoid most abbreviations)

Reply:
You could change the parameter name to `timoutDuration`

Reply:
That's much better. Done.